src/functions_linearalgebra.jl

```diff
@@ -1,32 +1,53 @@
 
 #==
     Note on implementation of factorisition types:
-    The general strategy is to make one of the fields of the factorization types
-    into a NamedDimsArray.
-    We can then dispatch on this as required, and strip it out with `parent` for accessing operations.
-    However, the type of the field does not actually always match to what it should be
-    from a mathematical perspective.
-    Which is corrected in `Base.getproperty`, as required
+    The general strategy is to create a named factorization type which wraps other factorizations.
+    This allows us to define `getproperty` methods which return `NamedDimsArray`s around the
+    corresponding properties of the parent factorization types.
+
+    NOTE: Maybe we could just have a `Named` type that works with arrays and factorizations
+    kind of like `Adjoint`?
 ==#
 
-## lu
+struct NamedFactorization{L, T, F<:Factorization{T}} <: Factorization{T}
+    fact::F
+end
+
+# Need parent to explicitly call `getfield` to avoid infinitely calling `getproperty`
+Base.parent(named::NamedFactorization) = getfield(named, :fact)
+Base.size(named::NamedFactorization) = size(parent(named))
+Base.propertynames(named::NamedFactorization; kwargs...) = propertynames(parent(named))
```

Comment:
It'd be nice if there was a `Factorization` test suite to check if my type fulfills the `Factorization` type appropriately?